test/PositionManager.ts

```diff
@@ -131,10 +191,201 @@ describe("GammaPoolFactory", function () {
+                expect(event.args.pool).to.equal(gammaPool.address);
+                expect(event.args.reservesLen).to.equal(3);
+                expect(event.args.assets.toNumber()).to.equal(17);
+            });
+        });
     });
+
     // You can nest describe calls to create subsections.
     describe("Long Gamma", function () {
+        it("createLoan", async function () {
+            await cfmm.approve(posMgr.address, ethers.constants.MaxUint256);//must approve before sending tokens
+            
+            console.log("event >>");
+            console.log(createLoanArgs);
+            expect(createLoanArgs.pool).to.equal(gammaPool.address);
+            expect(createLoanArgs.tokenId.toNumber()).to.equal(19);
+        });
 
+        it("borrowLiquidity", async function () {
+            await cfmm.approve(posMgr.address, ethers.constants.MaxUint256);//must approve before sending tokens
+
+            const BorrowLiquidityParams = {
+                cfmm: cfmm.address,
+                protocol: 1,
+                tokenId: tokenId,
+                lpTokens: 1,
+                to: owner.address,
+                deadline: ethers.constants.MaxUint256
+            }
+            
+            const res = await (await posMgr.borrowLiquidity(BorrowLiquidityParams)).wait();
+            
+            const { args } = res.events[0]
+            console.log("event >>");
+            console.log(args);
+            expect(args.pool).to.equal(gammaPool.address);
+            expect(args.tokenId.toNumber()).to.equal(19);
+        });
+
+        it("repayLiquidity", async function () {
+            await cfmm.approve(posMgr.address, ethers.constants.MaxUint256);//must approve before sending tokens
+            
+            const RepayLiquidityParams = {
+                cfmm: cfmm.address,
+                protocol: 1,
+                tokenId: tokenId,
+                liquidity: 1,
+                to: owner.address,
+                deadline: ethers.constants.MaxUint256
+            }
+            
+            const res = await (await posMgr.repayLiquidity(RepayLiquidityParams)).wait();
+            
+            const { args } = res.events[0]
+            console.log("event >>");
+            console.log(args);
+            expect(args.pool).to.equal(gammaPool.address);
+            expect(args.tokenId.toNumber()).to.equal(19);
+            expect(args.liquidityPaid.toNumber()).to.equal(24);
+            expect(args.lpTokensPaid.toNumber()).to.equal(25);
+            expect(args.amountsLen.toNumber()).to.equal(9);
+        });
+
+        it("increaseCollateral", async function () {
+            await tokenA.approve(posMgr.address, ethers.constants.MaxUint256);//must approve before sending tokens
+            await tokenB.approve(posMgr.address, ethers.constants.MaxUint256);//must approve before sending tokens
+            await cfmm.approve(posMgr.address, ethers.constants.MaxUint256);//must approve before sending tokens
```

Comment:
No need to approve cfmm here

Reply:
we are sending cfmm via AddRemoveCollateralParams right?

Reply:
yes, but you don't  really need to approve it. You can just pass the cfmm address in AddRemoveCollateralParams without approving